File: PayJunction C#/ConsoleApplication2/PayJunction/SmartTerminal/SmartTerminalTransactionRequest.cs
using RestSharp;
using Newtonsoft.Json;

namespace PayJunction.SmartTerminal
{
    class SmartTerminalTransactionRequest : Request<SmartTerminal.SmartTerminalTransactionResponse>
    {
        public override string EndPoint
        {
            get
            {
                return "smartterminals/" + SmartTerminalID + "/request-payment";
            }
        }

        public string SmartTerminalID { get; set; }

        public override Method Method
        {
            get
            {
                return Method.POST;
            }
        }

        public string AmountBase
        {
            get
            {
                return Params["amountBase"];
            }
            set
            {
                Params["amountBase"] = value;
            }
        }

        public string TerminalID
        {
            get
            {
                return Params["terminalId"];
            }
            set
            {
                Params["terminalId"] = value;
            }
        }

        public SmartTerminalTransactionRequest(string terminalID, string amountBase, string smartTerminalID)
        {
            this.TerminalID = terminalID;
            this.AmountBase = amountBase;
            this.SmartTerminalID = smartTerminalID;
        }

        public override SmartTerminalTransactionResponse ProcessResponse()
        {
            SmartTerminalTransactionResponse resp = JsonConvert.DeserializeObject<SmartTerminalTransactionResponse>(RawResponse);
            return resp;
        }
    }
}

File: PayJunction Webhook/PayJunction/PayJunction/SmartTerminal/SmartTerminalTransactionRequest.vb
Imports RestSharp
Imports Newtonsoft.Json

Namespace PayJunction.SmartTerminal
    Class SmartTerminalTransactionRequest
        Inherits Request(Of SmartTerminalTransactionResponse)
        Public Overrides ReadOnly Property EndPoint() As String
            Get
                Return (Convert.ToString("smartterminals/") & SmartTerminalID) + "/request-payment"
            End Get
        End Property

        Public Property SmartTerminalID() As String
            Get
                Return m_SmartTerminalID
            End Get
            Set
                m_SmartTerminalID = Value
            End Set
        End Property
        Private m_SmartTerminalID As String

        Public Overrides ReadOnly Property Method() As Method
            Get
                Return Method.POST
            End Get
        End Property

        Public Property AmountBase() As String
            Get
                Return Params("amountBase")
            End Get
            Set
                Params("amountBase") = Value
            End Set
        End Property

        Public Property TerminalID() As String
            Get
                Return Params("terminalId")
            End Get
            Set
                Params("terminalId") = Value
            End Set
        End Property

        Public Sub New(terminalID As String, amountBase As String, smartTerminalID As String)
            Me.TerminalID = terminalID
            Me.AmountBase = amountBase
            Me.SmartTerminalID = smartTerminalID
        End Sub

        Public Overrides Function ProcessResponse() As SmartTerminalTransactionResponse
            Dim resp As SmartTerminalTransactionResponse = JsonConvert.DeserializeObject(Of SmartTerminalTransactionResponse)(RawResponse)
            Return resp
        End Function
    End Class
End Namespace